File: incubating/sdk-client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/entity/ProtonClientOptions.kt
package me.proton.drive.sdk.entity

import me.proton.drive.sdk.LoggerProvider

data class ProtonClientOptions(
    val userAgent: String? = null,
    val baseUrl: String? = null,
    val bindingsLanguage: String? = null,
    val tlsPolicy: ProtonClientTlsPolicy? = null,
    val loggerProvider: LoggerProvider? = null,
    val entityCachePath: String? = null,
)

File: sdk-client/cs/sdk/src/Proton.Sdk/ProtonClientOptions.cs
using Proton.Sdk.Caching;
using Proton.Sdk.Http;
using Proton.Sdk.Telemetry;

namespace Proton.Sdk;

public record ProtonClientOptions
{
    public Uri? BaseUrl { get; set; }
    public string? UserAgent { get; set; }
    public ProtonClientTlsPolicy? TlsPolicy { get; set; }
    public Func<DelegatingHandler>? CustomHttpMessageHandlerFactory { get; set; }
    public IHttpClientFactory? HttpClientFactory { get; set; }
    public ICacheRepository? EntityCacheRepository { get; set; }
    public ITelemetry? Telemetry { get; set; }
    public IFeatureFlagProvider? FeatureFlagProvider { get; set; }
    internal ICacheRepository? SecretCacheRepository { get; set; }
    internal Uri? RefreshRedirectUri { get; set; }
    internal string? BindingsLanguage { get; set; }
}
